java/property/TypeProperty.java

```diff
@@ -27,17 +28,17 @@
  */
 public class TypeProperty extends VarProperty {
 
-    private final String name;
+    private final Label label;
 
-    public TypeProperty(String name) {
-        if (name == null) {
+    public TypeProperty(Label label) {
+        if (label == null) {
             throw new NullPointerException("Null label");
         }
-        this.name = name;
+        this.label = label;
     }
 
-    public String name() {
-        return name;
+    public Label label() {
+        return label;
```

Comment:
Have we considered not having to implement a `Label` class, and just have `TypeProperty` to have `label` and `scope`, where both are represented as Strings?

Reply:
yeah we need the class for the auto-generated parser, unless we want the auto-generated parser to return a Pair sometimes!

Reply:
Right